module-system/sov-modules-api/src/state/scratchpad.rs

```diff
@@ -268,6 +262,15 @@ impl<C: Context> WorkingSet<C> {
     pub fn charge_gas(&mut self, gas: &C::GasUnit) -> anyhow::Result<()> {
         self.gas_meter.charge_gas(gas)
     }
+
+    #[cfg(feature = "native")]
+    pub fn get_with_proof(
+        &mut self,
+        key: StorageKey,
+    ) -> sov_state::storage::StorageProof<<C::Storage as Storage>::Proof> {
+        // First inner is `RevertableWriter` and second inner is actually a `Storage` instance
+        self.delta.inner.inner.get_with_proof(key).unwrap()
```

Comment:
Can this unwrap panic?

Reply:
Yes, unfortunately. It will panic if underlying Storage implementation does not have implementation. For example if it called in Zk context by accident. I think native flag should protect against it.

Another option would be to have a trait definition to be different in `native` and `zk`, then we can get rid of the option